node/src/components/storage/lmdb_store.rs

```diff
@@ -34,6 +34,8 @@ impl<V: Value, M: Default + Send + Sync> LmdbStore<V, M> {
         let env = Environment::new()
             .set_flags(EnvironmentFlags::NO_SUB_DIR)
             .set_map_size(max_size)
+            // to avoid panic on excessive read-only transactions
+            .set_max_readers(MAX_THREAD_COUNT as u32)
```

Comment:
This will allow for `MAX_THREAD_COUNT` threads (currently 512) _per_ environment but we have more than one, right?

Reply:
I did test this by stress testing a testnet (not easy to say! :D) using nctl, and it does seem to be ok.

The tokio default is 512 threads and the LMDB default is 126 readers.